src/handlers/componentList.ts

```diff
@@ -41,6 +42,18 @@ export default async function componentList(): Promise<void> {
     );
   }
 
+  // Make sure the given system is installed and has the correct branch/commit/tag checked out.
+  try {
+    await cloneIntoCache('systems', [systemName])(emulsifyConfig.system);
```

Comment:
Ditto here, this will make sure the system exists within the cache before attempting to list components within the system/variant.